### PR TITLE
purge devitobench

### PR DESCRIPTION
**Pull request by mloubout**
See #1097

### CODE REVIEW

File: benchmarks/tools/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        return '_'.join(['%s%s' % p for p in params])
+
+    def lookup(self, params={}, event=None, measure='time', category='timings'):
+        """
+        Lookup a set of results accoridng to a parameter set.
+
+        Parameters
+        ----------
+        params : Dict
+            Parameter set by which to filter results
```
Comment (FabioLuporini) [inline benchmarks/tools/benchmark.py]: don't forget the full stops

File: benchmarks/tools/benchmark.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    help = 'Perform benchmarking runs on target machine'
+    parser_bench = subparsers.add_parser('bench', help=help)
+    help = 'Plot diagrams from stored results'
+    parser_plot = subparsers.add_parser('plot', help=help)
+    return parser, parser_bench, parser_plot
+
+
+class Benchmark(object):
+    """
+    Class storing performance data for a set of benchmark runs
```
Comment (FabioLuporini) [inline benchmarks/tools/benchmark.py]: nitpicking
in devito, docstrings never start with "class ...". 
Rather: "Performance data for a set of benchmark runs ...."

File: benchmarks/tools/executor.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+__all__ = ['Executor']
+
+
+class Executor(object):
+    """
+    Abstract container class for a single benchmark data point.
+
+    Parameters
+    ----------
+    # TOFIX?
```
Comment (FabioLuporini) [inline benchmarks/tools/executor.py]: what?

Comment (georgebisbas) [reply benchmarks/tools/executor.py]: Is the comm fine like this? Just that.

File: benchmarks/tools/plotter.py [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+except ImportError:
+    b2m = None
+
+
+__all__ = ['Plotter', 'LinePlotter', 'RooflinePlotter', 'BarchartPlotter',
+           'AxisScale']
+
+
+def scale_limits(minval, maxval, base, type='log'):
+    """
```
Comment (georgebisbas) [inline benchmarks/tools/plotter.py]: Needs work

File: benchmarks/tools/plotter.py [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+        return dtype(base) * basevals
+
+
+class AxisScale(object):
+    """
+    Utility class to describe and configure axis value labelling.
+    """
+    def __init__(self, scale='log', base=2., dtype=np.float32,
+                 minval=None, maxval=None):
+        self.scale = scale
```
Comment (georgebisbas) [inline benchmarks/tools/plotter.py]: needs work

File: benchmarks/tools/benchmark.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
+from argparse import ArgumentParser
+from collections import OrderedDict
+from collections.abc import Iterable
+from itertools import product
+from datetime import datetime
+from os import path, makedirs
+from devito.logger import warning
```
Comment (FabioLuporini) [inline benchmarks/tools/benchmark.py]: blank line missing before the devito-related imports

File: benchmarks/tools/benchmark.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+import json
```
Comment (FabioLuporini) [inline benchmarks/tools/benchmark.py]: is json part of the standard library ? if so, it should appear in the group above?

File: benchmarks/tools/benchmark.py (hunk 6 above)
Comment (georgebisbas) [reply benchmarks/tools/benchmark.py]: fixed

File: benchmarks/tools/benchmark.py (hunk 7 above)
Comment (georgebisbas) [reply benchmarks/tools/benchmark.py]: fixed

File: benchmarks/user/benchmark.py [hunk 8]
```diff
@@ -314,7 +317,6 @@ def plot(problem, **kwargs):
                 % (problem, model_type[problem]['default-section']))
         section = model_type[problem]['default-section']
 
-    RooflinePlotter = get_ob_plotter()
```
Comment (FabioLuporini) [inline benchmarks/user/benchmark.py]: what was this??

Comment (FabioLuporini) [reply benchmarks/user/benchmark.py]: lolz, a leftover from God knows what ?

File: benchmarks/user/tools/executor.py [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        raise NotImplementedError("No custom executor function specified")
+
+    def register(self, value, event='execute', measure='time'):
+        """
+        Register a single timing value for a given event key.
+
+        Parameters
+        ----------
+        event : str
+            key for the measured event, ie. 'assembly' or 'solve'
```
Comment (georgebisbas) [inline benchmarks/user/tools/executor.py]: dots needed.

File: benchmarks/user/tools/plotter.py [hunk 10]
```diff
@@ -0,0 +1,506 @@
...
+except ImportError:
+    b2m = None
+
+
+__all__ = ['Plotter', 'LinePlotter', 'RooflinePlotter', 'BarchartPlotter',
+           'AxisScale']
+
+
+def scale_limits(minval, maxval, base, type='log'):
+
```
Comment (georgebisbas) [inline benchmarks/user/tools/plotter.py]: Delete line?

Comment (georgebisbas) [reply benchmarks/user/tools/plotter.py]: and add paramters

File: benchmarks/user/tools/plotter.py [hunk 11]
```diff
@@ -0,0 +1,506 @@
...
+        if not path.exists(self.plotdir):
+            makedirs(self.plotdir)
+        figpath = path.join(self.plotdir, figname)
+        info("Plotting %s " % figpath)
+        figure.savefig(figpath, format='pdf', facecolor='white',
+                       orientation='landscape', bbox_inches='tight')
+
+
+class LinePlotter(Plotter):
+
```
Comment (georgebisbas) [inline benchmarks/user/tools/plotter.py]: same

File: benchmarks/user/tools/plotter.py [hunk 12]
```diff
@@ -0,0 +1,506 @@
...
+        # Record all points keyed by group and legend labels
+        self.values[grouplabel][label] = value
+
+        # Record legend labels to avoid replication
+        if label is not None:
+            self.legend[label] = color
+
+
+class RooflinePlotter(Plotter):
+
```
Comment (georgebisbas) [inline benchmarks/user/tools/plotter.py]: same

**Pull request merged**